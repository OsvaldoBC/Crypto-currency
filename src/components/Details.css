.details-main {
  display: flex;
  justify-content: space-around;
  background-color: #4369b2;
  color: #fff;
  padding: 40px;
  gap: 40px;
}

.details-main img {
  width: 100px;
  height: 100px;
}

.details-metrics {
  background-color: #000;
  color: #fff;
  height: 100vh;
  display: grid;
  grid-template-columns: repeat(2, minmax(0, 1fr));
  gap: 30px;
  width: 100%;
}

.icon-metric {
  display: flex;
  flex-direction: column;
  padding-top: 20px;
  align-items: center;
  gap: 20px;
}
