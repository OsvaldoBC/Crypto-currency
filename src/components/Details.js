import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faMoneyCheckDollar, faChartLine, faGlobe, faClock, faCalendarDay, faCalendarWeek,
}
  from '@fortawesome/free-solid-svg-icons';
import { fetchDetails } from '../redux/CryptoDetails';
import GoingBack from './GoingBack';
import './Details.css';

const Details = () => {
  const { id } = useParams();
  const dispatch = useDispatch();

  const { detail, loading, error } = useSelector((state) => state.detail);

  useEffect(() => {
    dispatch(fetchDetails(id));
  }, [dispatch, id]);

  return (
    <div>
      {loading && <h1 className="fallback">Loading...</h1>}
      {error && <h1 className="fallback">{error}</h1>}
      {detail && (
        <div>
          <GoingBack />
          <div>
            <div className="details-main" key={detail.id}>
              <img src={detail.icon} alt={detail.id} />
              <div className="details-title">
                <h2>{detail.name}</h2>
                <h4>
                  <span> Symbol:</span>
                  {' '}
                  {detail.symbol}
                </h4>
                <h4>
                  <span> Rank:</span>
                  {' '}
                  {detail.rank}
                </h4>
                <h4>
                  <span> Price:</span>
                  {' '}
                  {detail.price}
                  {' '}
                  $
                </h4>
              </div>
            </div>
            <div className="details-metrics">
              <div className="icon-metric">
                <FontAwesomeIcon icon={faMoneyCheckDollar} size="4x" />
                <h3>
                  <span> Price in BTC:</span>
                  {' '}
                  {detail.priceBtc}
                </h3>
              </div>
              <div className="icon-metric">
                <FontAwesomeIcon icon={faChartLine} size="4x" />
                <h3>
                  <span>Available Supply:</span>
                  {' '}
                  {detail.availableSupply}
                </h3>
              </div>
              <div className="icon-metric">
                <FontAwesomeIcon icon={faGlobe} size="4x" />
                <h3>
                  <span>Total Supply:</span>
                  {' '}
                  {detail.totalSupply}
                </h3>
              </div>
              <div className="icon-metric">
                <FontAwesomeIcon icon={faClock} size="4x" />
                <h3>
                  <span>Price Change in hour:</span>
                  {' '}
                  {detail.priceChange1h}
                </h3>
              </div>
              <div className="icon-metric">
                <FontAwesomeIcon icon={faCalendarDay} size="4x" />
                <h3>
                  <span>Price Change in day:</span>
                  {' '}
                  {detail.priceChange1d}
                </h3>
              </div>
              <div className="icon-metric">
                <FontAwesomeIcon icon={faCalendarWeek} size="4x" />
                <h3>
                  <span>Price Change in week:</span>
                  {' '}
                  {detail.priceChange1w}
                </h3>
              </div>
            </div>
          </div>
        </div>
      )}
    </div>
  );
};
export default Details;
