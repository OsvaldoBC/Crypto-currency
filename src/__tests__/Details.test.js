import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import React from 'react';
import store from '../redux/configureStore';
import Details from '../components/Details';

describe('Details component', () => {
  it('Render Details component', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Details />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});
