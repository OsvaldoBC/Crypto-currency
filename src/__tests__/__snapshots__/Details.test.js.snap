// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component Render Details component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1
          class="fallback"
        >
          Loading...
        </h1>
        <div>
          <div
            class="back-container"
          >
            <button
              class="back-btn"
              type="button"
            >
              <h3>
                Go Back
              </h3>
            </button>
          </div>
          <div>
            <div
              class="details-main"
            >
              <img />
              <div
                class="details-title"
              >
                <h2 />
                <h4>
                  <span>
                     Symbol:
                  </span>
                   
                </h4>
                <h4>
                  <span>
                     Rank:
                  </span>
                   
                </h4>
                <h4>
                  <span>
                     Price:
                  </span>
                   
                   
                  $
                </h4>
              </div>
            </div>
            <div
              class="details-metrics"
            >
              <div
                class="icon-metric"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-money-check-dollar fa-4x "
                  data-icon="money-check-dollar"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 576 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M64 64C28.7 64 0 92.7 0 128V384c0 35.3 28.7 64 64 64H512c35.3 0 64-28.7 64-64V128c0-35.3-28.7-64-64-64H64zM272 192H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H272c-8.8 0-16-7.2-16-16s7.2-16 16-16zM256 304c0-8.8 7.2-16 16-16H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H272c-8.8 0-16-7.2-16-16zM164 152v13.9c7.5 1.2 14.6 2.9 21.1 4.7c10.7 2.8 17 13.8 14.2 24.5s-13.8 17-24.5 14.2c-11-2.9-21.6-5-31.2-5.2c-7.9-.1-16 1.8-21.5 5c-4.8 2.8-6.2 5.6-6.2 9.3c0 1.8 .1 3.5 5.3 6.7c6.3 3.8 15.5 6.7 28.3 10.5l.7 .2c11.2 3.4 25.6 7.7 37.1 15c12.9 8.1 24.3 21.3 24.6 41.6c.3 20.9-10.5 36.1-24.8 45c-7.2 4.5-15.2 7.3-23.2 9V360c0 11-9 20-20 20s-20-9-20-20V345.4c-10.3-2.2-20-5.5-28.2-8.4l0 0 0 0c-2.1-.7-4.1-1.4-6.1-2.1c-10.5-3.5-16.1-14.8-12.6-25.3s14.8-16.1 25.3-12.6c2.5 .8 4.9 1.7 7.2 2.4c13.6 4.6 24 8.1 35.1 8.5c8.6 .3 16.5-1.6 21.4-4.7c4.1-2.5 6-5.5 5.9-10.5c0-2.9-.8-5-5.9-8.2c-6.3-4-15.4-6.9-28-10.7l-1.7-.5c-10.9-3.3-24.6-7.4-35.6-14c-12.7-7.7-24.6-20.5-24.7-40.7c-.1-21.1 11.8-35.7 25.8-43.9c6.9-4.1 14.5-6.8 22.2-8.5V152c0-11 9-20 20-20s20 9 20 20z"
                    fill="currentColor"
                  />
                </svg>
                <h3>
                  <span>
                     Price in BTC:
                  </span>
                   
                </h3>
              </div>
              <div
                class="icon-metric"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-chart-line fa-4x "
                  data-icon="chart-line"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M64 64c0-17.7-14.3-32-32-32S0 46.3 0 64V400c0 44.2 35.8 80 80 80H480c17.7 0 32-14.3 32-32s-14.3-32-32-32H80c-8.8 0-16-7.2-16-16V64zm406.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L320 210.7l-57.4-57.4c-12.5-12.5-32.8-12.5-45.3 0l-112 112c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L240 221.3l57.4 57.4c12.5 12.5 32.8 12.5 45.3 0l128-128z"
                    fill="currentColor"
                  />
                </svg>
                <h3>
                  <span>
                    Available Supply:
                  </span>
                   
                </h3>
              </div>
              <div
                class="icon-metric"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-globe fa-4x "
                  data-icon="globe"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M352 256c0 22.2-1.2 43.6-3.3 64H163.3c-2.2-20.4-3.3-41.8-3.3-64s1.2-43.6 3.3-64H348.7c2.2 20.4 3.3 41.8 3.3 64zm28.8-64H503.9c5.3 20.5 8.1 41.9 8.1 64s-2.8 43.5-8.1 64H380.8c2.1-20.6 3.2-42 3.2-64s-1.1-43.4-3.2-64zm112.6-32H376.7c-10-63.9-29.8-117.4-55.3-151.6c78.3 20.7 142 77.5 171.9 151.6zm-149.1 0H167.7c6.1-36.4 15.5-68.6 27-94.7c10.5-23.6 22.2-40.7 33.5-51.5C239.4 3.2 248.7 0 256 0s16.6 3.2 27.8 13.8c11.3 10.8 23 27.9 33.5 51.5c11.6 26 21 58.2 27 94.7zm-209 0H18.6C48.6 85.9 112.2 29.1 190.6 8.4C165.1 42.6 145.3 96.1 135.3 160zM8.1 192H131.2c-2.1 20.6-3.2 42-3.2 64s1.1 43.4 3.2 64H8.1C2.8 299.5 0 278.1 0 256s2.8-43.5 8.1-64zM194.7 446.6c-11.6-26-20.9-58.2-27-94.6H344.3c-6.1 36.4-15.5 68.6-27 94.6c-10.5 23.6-22.2 40.7-33.5 51.5C272.6 508.8 263.3 512 256 512s-16.6-3.2-27.8-13.8c-11.3-10.8-23-27.9-33.5-51.5zM135.3 352c10 63.9 29.8 117.4 55.3 151.6C112.2 482.9 48.6 426.1 18.6 352H135.3zm358.1 0c-30 74.1-93.6 130.9-171.9 151.6c25.5-34.2 45.2-87.7 55.3-151.6H493.4z"
                    fill="currentColor"
                  />
                </svg>
                <h3>
                  <span>
                    Total Supply:
                  </span>
                   
                </h3>
              </div>
              <div
                class="icon-metric"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-clock fa-4x "
                  data-icon="clock"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 0a256 256 0 1 1 0 512A256 256 0 1 1 256 0zM232 120V256c0 8 4 15.5 10.7 20l96 64c11 7.4 25.9 4.4 33.3-6.7s4.4-25.9-6.7-33.3L280 243.2V120c0-13.3-10.7-24-24-24s-24 10.7-24 24z"
                    fill="currentColor"
                  />
                </svg>
                <h3>
                  <span>
                    Price Change in hour:
                  </span>
                   
                </h3>
              </div>
              <div
                class="icon-metric"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-calendar-day fa-4x "
                  data-icon="calendar-day"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M128 0c17.7 0 32 14.3 32 32V64H288V32c0-17.7 14.3-32 32-32s32 14.3 32 32V64h48c26.5 0 48 21.5 48 48v48H0V112C0 85.5 21.5 64 48 64H96V32c0-17.7 14.3-32 32-32zM0 192H448V464c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V192zm80 64c-8.8 0-16 7.2-16 16v96c0 8.8 7.2 16 16 16h96c8.8 0 16-7.2 16-16V272c0-8.8-7.2-16-16-16H80z"
                    fill="currentColor"
                  />
                </svg>
                <h3>
                  <span>
                    Price Change in day:
                  </span>
                   
                </h3>
              </div>
              <div
                class="icon-metric"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-calendar-week fa-4x "
                  data-icon="calendar-week"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M128 0c17.7 0 32 14.3 32 32V64H288V32c0-17.7 14.3-32 32-32s32 14.3 32 32V64h48c26.5 0 48 21.5 48 48v48H0V112C0 85.5 21.5 64 48 64H96V32c0-17.7 14.3-32 32-32zM0 192H448V464c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V192zm80 64c-8.8 0-16 7.2-16 16v64c0 8.8 7.2 16 16 16H368c8.8 0 16-7.2 16-16V272c0-8.8-7.2-16-16-16H80z"
                    fill="currentColor"
                  />
                </svg>
                <h3>
                  <span>
                    Price Change in week:
                  </span>
                   
                </h3>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1
        class="fallback"
      >
        Loading...
      </h1>
      <div>
        <div
          class="back-container"
        >
          <button
            class="back-btn"
            type="button"
          >
            <h3>
              Go Back
            </h3>
          </button>
        </div>
        <div>
          <div
            class="details-main"
          >
            <img />
            <div
              class="details-title"
            >
              <h2 />
              <h4>
                <span>
                   Symbol:
                </span>
                 
              </h4>
              <h4>
                <span>
                   Rank:
                </span>
                 
              </h4>
              <h4>
                <span>
                   Price:
                </span>
                 
                 
                $
              </h4>
            </div>
          </div>
          <div
            class="details-metrics"
          >
            <div
              class="icon-metric"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-money-check-dollar fa-4x "
                data-icon="money-check-dollar"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M64 64C28.7 64 0 92.7 0 128V384c0 35.3 28.7 64 64 64H512c35.3 0 64-28.7 64-64V128c0-35.3-28.7-64-64-64H64zM272 192H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H272c-8.8 0-16-7.2-16-16s7.2-16 16-16zM256 304c0-8.8 7.2-16 16-16H496c8.8 0 16 7.2 16 16s-7.2 16-16 16H272c-8.8 0-16-7.2-16-16zM164 152v13.9c7.5 1.2 14.6 2.9 21.1 4.7c10.7 2.8 17 13.8 14.2 24.5s-13.8 17-24.5 14.2c-11-2.9-21.6-5-31.2-5.2c-7.9-.1-16 1.8-21.5 5c-4.8 2.8-6.2 5.6-6.2 9.3c0 1.8 .1 3.5 5.3 6.7c6.3 3.8 15.5 6.7 28.3 10.5l.7 .2c11.2 3.4 25.6 7.7 37.1 15c12.9 8.1 24.3 21.3 24.6 41.6c.3 20.9-10.5 36.1-24.8 45c-7.2 4.5-15.2 7.3-23.2 9V360c0 11-9 20-20 20s-20-9-20-20V345.4c-10.3-2.2-20-5.5-28.2-8.4l0 0 0 0c-2.1-.7-4.1-1.4-6.1-2.1c-10.5-3.5-16.1-14.8-12.6-25.3s14.8-16.1 25.3-12.6c2.5 .8 4.9 1.7 7.2 2.4c13.6 4.6 24 8.1 35.1 8.5c8.6 .3 16.5-1.6 21.4-4.7c4.1-2.5 6-5.5 5.9-10.5c0-2.9-.8-5-5.9-8.2c-6.3-4-15.4-6.9-28-10.7l-1.7-.5c-10.9-3.3-24.6-7.4-35.6-14c-12.7-7.7-24.6-20.5-24.7-40.7c-.1-21.1 11.8-35.7 25.8-43.9c6.9-4.1 14.5-6.8 22.2-8.5V152c0-11 9-20 20-20s20 9 20 20z"
                  fill="currentColor"
                />
              </svg>
              <h3>
                <span>
                   Price in BTC:
                </span>
                 
              </h3>
            </div>
            <div
              class="icon-metric"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chart-line fa-4x "
                data-icon="chart-line"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M64 64c0-17.7-14.3-32-32-32S0 46.3 0 64V400c0 44.2 35.8 80 80 80H480c17.7 0 32-14.3 32-32s-14.3-32-32-32H80c-8.8 0-16-7.2-16-16V64zm406.6 86.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L320 210.7l-57.4-57.4c-12.5-12.5-32.8-12.5-45.3 0l-112 112c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L240 221.3l57.4 57.4c12.5 12.5 32.8 12.5 45.3 0l128-128z"
                  fill="currentColor"
                />
              </svg>
              <h3>
                <span>
                  Available Supply:
                </span>
                 
              </h3>
            </div>
            <div
              class="icon-metric"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-globe fa-4x "
                data-icon="globe"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M352 256c0 22.2-1.2 43.6-3.3 64H163.3c-2.2-20.4-3.3-41.8-3.3-64s1.2-43.6 3.3-64H348.7c2.2 20.4 3.3 41.8 3.3 64zm28.8-64H503.9c5.3 20.5 8.1 41.9 8.1 64s-2.8 43.5-8.1 64H380.8c2.1-20.6 3.2-42 3.2-64s-1.1-43.4-3.2-64zm112.6-32H376.7c-10-63.9-29.8-117.4-55.3-151.6c78.3 20.7 142 77.5 171.9 151.6zm-149.1 0H167.7c6.1-36.4 15.5-68.6 27-94.7c10.5-23.6 22.2-40.7 33.5-51.5C239.4 3.2 248.7 0 256 0s16.6 3.2 27.8 13.8c11.3 10.8 23 27.9 33.5 51.5c11.6 26 21 58.2 27 94.7zm-209 0H18.6C48.6 85.9 112.2 29.1 190.6 8.4C165.1 42.6 145.3 96.1 135.3 160zM8.1 192H131.2c-2.1 20.6-3.2 42-3.2 64s1.1 43.4 3.2 64H8.1C2.8 299.5 0 278.1 0 256s2.8-43.5 8.1-64zM194.7 446.6c-11.6-26-20.9-58.2-27-94.6H344.3c-6.1 36.4-15.5 68.6-27 94.6c-10.5 23.6-22.2 40.7-33.5 51.5C272.6 508.8 263.3 512 256 512s-16.6-3.2-27.8-13.8c-11.3-10.8-23-27.9-33.5-51.5zM135.3 352c10 63.9 29.8 117.4 55.3 151.6C112.2 482.9 48.6 426.1 18.6 352H135.3zm358.1 0c-30 74.1-93.6 130.9-171.9 151.6c25.5-34.2 45.2-87.7 55.3-151.6H493.4z"
                  fill="currentColor"
                />
              </svg>
              <h3>
                <span>
                  Total Supply:
                </span>
                 
              </h3>
            </div>
            <div
              class="icon-metric"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-clock fa-4x "
                data-icon="clock"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 0a256 256 0 1 1 0 512A256 256 0 1 1 256 0zM232 120V256c0 8 4 15.5 10.7 20l96 64c11 7.4 25.9 4.4 33.3-6.7s4.4-25.9-6.7-33.3L280 243.2V120c0-13.3-10.7-24-24-24s-24 10.7-24 24z"
                  fill="currentColor"
                />
              </svg>
              <h3>
                <span>
                  Price Change in hour:
                </span>
                 
              </h3>
            </div>
            <div
              class="icon-metric"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-calendar-day fa-4x "
                data-icon="calendar-day"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 0c17.7 0 32 14.3 32 32V64H288V32c0-17.7 14.3-32 32-32s32 14.3 32 32V64h48c26.5 0 48 21.5 48 48v48H0V112C0 85.5 21.5 64 48 64H96V32c0-17.7 14.3-32 32-32zM0 192H448V464c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V192zm80 64c-8.8 0-16 7.2-16 16v96c0 8.8 7.2 16 16 16h96c8.8 0 16-7.2 16-16V272c0-8.8-7.2-16-16-16H80z"
                  fill="currentColor"
                />
              </svg>
              <h3>
                <span>
                  Price Change in day:
                </span>
                 
              </h3>
            </div>
            <div
              class="icon-metric"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-calendar-week fa-4x "
                data-icon="calendar-week"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 0c17.7 0 32 14.3 32 32V64H288V32c0-17.7 14.3-32 32-32s32 14.3 32 32V64h48c26.5 0 48 21.5 48 48v48H0V112C0 85.5 21.5 64 48 64H96V32c0-17.7 14.3-32 32-32zM0 192H448V464c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V192zm80 64c-8.8 0-16 7.2-16 16v64c0 8.8 7.2 16 16 16H368c8.8 0 16-7.2 16-16V272c0-8.8-7.2-16-16-16H80z"
                  fill="currentColor"
                />
              </svg>
              <h3>
                <span>
                  Price Change in week:
                </span>
                 
              </h3>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
